In [5]:
file = open("cryptology.txt", "w")

for i in range(5):
    line = input("Enter statement " + str(i + 1) + ": ")
    file.write(line + "\n")

file.close()

print("Data stored successfully.")

Enter statement 1:  Cryptology is the study of secure communication.
Enter statement 2:  Cryptography focuses on protecting information.
Enter statement 3:  Plaintext represents the original message.
Enter statement 4:  Ciphertext represents the transformed message.
Enter statement 5:  Cryptanalysis is fun :)  I love Cryptanalysis ;)


Data stored successfully.


In [6]:
file = open("cryptology.txt", "r")
content = file.read()
print(content)
file.close()

Cryptology is the study of secure communication.
Cryptography focuses on protecting information.
Plaintext represents the original message.
Ciphertext represents the transformed message.
Cryptanalysis is fun :)  I love Cryptanalysis ;)



3. Write a program to read cryptology.txt one line at a time and display each line along with its line number.


In [7]:
file = open("cryptology.txt", "r")
line_number = 1

for line in file:
    print(line_number, line, end="")
    line_number += 1

file.close()

1 Cryptology is the study of secure communication.
2 Cryptography focuses on protecting information.
3 Plaintext represents the original message.
4 Ciphertext represents the transformed message.
5 Cryptanalysis is fun :)  I love Cryptanalysis ;)


4. write a program to store the output in a file called cryptology_no.txt


In [10]:
input_file = open("cryptology.txt", "r")
output_file = open("cryptology_no.txt", "w")
line_number = 1

for line in input_file:
    output_file.write(str(line_number) + " " + line)
    line_number += 1

input_file.close()
output_file.close()

print("Output stored in cryptology_no.txt")

Output stored in cryptology_no.txt


5. Write a program to read cryptology.txt and determine:

Number of lines
Number of words
Number of characters

In [11]:
file = open("cryptology.txt", "r")
content = file.read()
lines = content.splitlines()
words = content.split()
characters = len(content)

print("Number of lines :", len(lines))
print("Number of words :", len(words))
print("Number of characters :", characters)

file.close()

Number of lines : 5
Number of words : 30
Number of characters : 236


6. Read cryptology_no.txt and count the number of:

Uppercase letters
Lowercase letters
Digits
Spaces
Special characters

In [12]:
file = open("cryptology_no.txt", "r")
content = file.read()
uppercase = 0
lowercase = 0
digits = 0
spaces = 0
special = 0

for ch in content:
    if ch.isupper():
        uppercase += 1
    elif ch.islower():
        lowercase += 1
    elif ch.isdigit():
        digits += 1
    elif ch == " ":
        spaces += 1
    elif ch != "\n":
        special += 1

print("Uppercase letters :", uppercase)
print("Lowercase letters :", lowercase)
print("Digits            :", digits)
print("Spaces            :", spaces)
print("Special characters:", special)

file.close()

Uppercase letters : 7
Lowercase letters : 190
Digits            : 5
Spaces            : 31
Special characters: 8


7. Write a program that accepts a word from the user and searches for that word in cryptology.txt.


In [13]:
word = input("Enter word to search: ")
file = open("cryptology.txt", "r")
found = False

for line in file:
    if word.lower() in line.lower():
        print("Word found in:", line, end="")
        found = True

if not found:
    print("Word not found.")

file.close()

Enter word to search:  love


Word found in: Cryptanalysis is fun :)  I love Cryptanalysis ;)


8. Read cryptology.txt and display only the lines containing the word message.


In [14]:
file = open("cryptology.txt", "r")
for line in file:
    if "message" in line.lower():
        print(line, end="")

file.close()

Plaintext represents the original message.
Ciphertext represents the transformed message.


9. Write a Python program to read cryptology.txt, convert all alphabetic characters to uppercase, and store the result in a new file called normalized.txt.


In [16]:
in_file = open("cryptology.txt", "r")
out_file = open("normalized.txt", "w")

for line in in_file:
    out_file.write(line.upper())

in_file.close()
out_file.close()
print("Normalized file created successfully.")

Normalized file created successfully.


10. Write a program to read the file and print only the lines containing only alphabetic characters from the cryptology_no.txt file.


In [20]:
file = open("cryptology_no.txt", "r")

for line in file:
    part = line.split(" ", 1)

    if len(part) == 2:
        text = part[1].strip()

        if text.isalpha():
            print(text)

file.close()

11. Write  a program to get the string from the user and reverse the string. 


In [21]:
string = input("Enter a string: ")
reverse = string[::-1]

print("Reversed string:", reverse)

Enter a string:  Improve the above code to reverse the string from the file


Reversed string: elif eht morf gnirts eht esrever ot edoc evoba eht evorpmI


In [ ]:
12. Improve the above code to reverse the string from the file.


In [22]:
file = open("cryptology.txt", "r")

for line in file:
    line = line.rstrip("\n")
    print(line[::-1])

file.close()

.noitacinummoc eruces fo yduts eht si ygolotpyrC
.noitamrofni gnitcetorp no sesucof yhpargotpyrC
.egassem lanigiro eht stneserper txetnialP
.egassem demrofsnart eht stneserper txetrehpiC
); sisylanatpyrC evol I  ): nuf si sisylanatpyrC


13. Simulate the tac command in Python. 


In [23]:
file = open("cryptology.txt", "r")

lines = file.readlines()

for line in reversed(lines):
    print(line, end="")

file.close()

Cryptanalysis is fun :)  I love Cryptanalysis ;)
Ciphertext represents the transformed message.
Plaintext represents the original message.
Cryptography focuses on protecting information.
Cryptology is the study of secure communication.


In [ ]:
Create a Python program that accepts five lines of text related to cryptology from the user and stores them in input.txt. Read the file and determine the number of lines, words, alphabetic characters, digits, spaces, and special characters. Convert the text to uppercase and store the processed contents in processed.txt. Finally, read and display the contents of processed.txt.


In [24]:
file = open("input.txt", "w")

for i in range(5):
    line = input("Enter line " + str(i + 1) + ": ")
    file.write(line + "\n")

file.close()

file = open("input.txt", "r")

content = file.read()
# counting
lines = content.splitlines()
words = content.split()

line_count = len(lines)
word_count = len(words)

alphabetic = 0
digits = 0
spaces = 0
special = 0

for ch in content:
    if ch.isalpha():
        alphabetic += 1
    elif ch.isdigit():
        digits += 1
    elif ch == " ":
        spaces += 1
    elif ch != "\n":
        special += 1

file.close()

# display
print("Number of lines :", line_count)
print("Number of words :", word_count)
print("Alphabetic characters :", alphabetic)
print("Digits :", digits)
print("Spaces :", spaces)
print("Special characters :", special)

# to upper
file = open("processed.txt", "w")

file.write(content.upper())

file.close()

file = open("processed.txt", "r")

print("\n----- Processed Contents -----")
print(file.read())

file.close()

Enter line 1:  Cryptology is the study of secure communication. 
Enter line 2:  Cryptanalysis focuses on analyzing cryptographic systems.
Enter line 3:  Plaintext represents the original message.
Enter line 4:  Ciphertext represents the transformed message.
Enter line 5:  Cryptography protects information.


Number of lines : 5
Number of words : 26
Alphabetic characters : 201
Digits : 0
Spaces : 22
Special characters : 5

----- Processed Contents -----
CRYPTOLOGY IS THE STUDY OF SECURE COMMUNICATION. 
CRYPTANALYSIS FOCUSES ON ANALYZING CRYPTOGRAPHIC SYSTEMS.
PLAINTEXT REPRESENTS THE ORIGINAL MESSAGE.
CIPHERTEXT REPRESENTS THE TRANSFORMED MESSAGE.
CRYPTOGRAPHY PROTECTS INFORMATION.



In [25]:
import re

file = open("security_data.txt", "r")
content = file.read()

pattern = r'\b(?:\d{1,3}\.){3}\d{1,3}\b'

ips = re.findall(pattern, content)

print("IPv4-address-like patterns:")
for ip in ips:
    print(ip)

file.close()

IPv4-address-like patterns:
192.168.10.15
10.10.5.21


2. Write a Python program to identify and display all email-address-like patterns present in security_data.txt.



In [26]:
import re

file = open("security_data.txt", "r")
content = file.read()

pattern = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b'

emails = re.findall(pattern, content)

print("Email addresses:")
for email in emails:
    print(email)

file.close()

Email addresses:
admin@securitylab.com


3. Write a Python program to extract session IDs that follow the format SEC-YYYY-NNNN, where YYYY and NNNN represent four digits.


In [27]:
import re

file = open("security_data.txt", "r")
content = file.read()

pattern = r'\bSEC-\d{4}-\d{4}\b'

session_ids = re.findall(pattern, content)

print("Session IDs:")
for session in session_ids:
    print(session)

file.close()

Session IDs:
SEC-2026-1045


In [28]:
import re

file = open("security_data.txt", "r")
content = file.read()

pattern = r'\b[0-9a-fA-F]+\b'

hex_values = re.findall(pattern, content)

print("Hexadecimal sequences:")
for value in hex_values:
    print(value)

file.close()

Hexadecimal sequences:
192
168
10
15
10
10
5
21
5
2026
1045
a94f3c2d8e71b05f
443


5. Write a Python program to read security_data.txt and display all lines containing the word failed, irrespective of uppercase or lowercase.

In [29]:
import re

file = open("security_data.txt", "r")

for line in file:
    if re.search(r'\bfailed\b', line, re.IGNORECASE):
        print(line, end="")

file.close()

Failed login attempts: 5
User admin failed authentication

6. Write a Python program to identify and extract the port number from statements of the form Port 443 connection established.


In [30]:
import re

file = open("security_data.txt", "r")
content = file.read()

pattern = r'\bPort\s+(\d+)\s+connection established\b'

match = re.search(pattern, content)

if match:
    print("Port number:", match.group(1))
else:
    print("Port number not found.")

file.close()

Port number: 443


7.Write a Python program to read the contents of security_data.txt and use regular expressions to extract IP-address-like patterns, email addresses, session IDs, and hexadecimal sequences. Store the identified patterns in a new file named pattern_results.txt.


In [32]:
import re

# Read security_data.txt
file = open("security_data.txt", "r")
content = file.read()
file.close()

# Regular expressions
ip_pattern = r'\b(?:\d{1,3}\.){3}\d{1,3}\b'
email_pattern = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b'
session_pattern = r'\bSEC-\d{4}-\d{4}\b'
hex_pattern = r'\b[0-9a-fA-F]{8,}\b'

# Extract patterns
ips = re.findall(ip_pattern, content)
emails = re.findall(email_pattern, content)
sessions = re.findall(session_pattern, content)
hex_values = re.findall(hex_pattern, content)

# Store results
file = open("pattern_results.txt", "w")

file.write("IPv4 Addresses:\n")
for ip in ips:
    file.write(ip + "\n")

file.write("\nEmail Addresses:\n")
for email in emails:
    file.write(email + "\n")

file.write("\nSession IDs:\n")
for session in sessions:
    file.write(session + "\n")

file.write("\nHexadecimal Sequences:\n")
for value in hex_values:
    file.write(value + "\n")

file.close()

print("Patterns extracted and stored in pattern_results.txt")

Patterns extracted and stored in pattern_results.txt
